frontend/src/pages/analytics/segments/components/group/ConditionGroup.tsx

```diff
@@ -0,0 +1,263 @@
+ *  and limitations under the License.
+ */
+
+import { Button } from '@cloudscape-design/components';
+import {
+  ERelationShip,
+  IEventSegmentationItem,
+  INIT_SEGMENTATION_DATA,
+} from 'components/eventselect/AnalyticsType';
+import AnalyticsSegmentFilter from 'components/eventselect/reducer/AnalyticsSegmentFilter';
+import { analyticsSegmentFilterReducer } from 'components/eventselect/reducer/analyticsSegmentFilterReducer';
+import { AnalyticsSegmentActionType } from 'components/eventselect/reducer/analyticsSegmentGroupReducer';
+import { useSegmentContext } from 'context/SegmentContext';
+import { identity } from 'lodash';
+import React, { useMemo, useReducer, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { ConditionType } from 'ts/const';
+import Condition from './Condition';
+import EventSeqItem from './EventSeqItem';
+import UserDoneComp from './type/UserDoneComp';
+import UserDoneInSeq from './type/UserDoneInSeq';
+import UserInGroup from './type/UserInGroup';
+import UserIsComp from './type/UserIsComp';
+
+export interface SegmentPropsData {
+  level: number;
+  rootIndex: number;
+  parentIndex: number;
+  currentIndex: number;
+  parentData: IEventSegmentationItem;
+  sequenceEventIndex?: number;
+  sequenceConditionIndex?: number;
+}
+
+interface ConditionGroupProps {
+  segmentData: IEventSegmentationItem;
+  segmentProps: SegmentPropsData;
+}
+
+const ConditionGroup: React.FC<ConditionGroupProps> = (
+  props: ConditionGroupProps
+) => {
+  const { t } = useTranslation();
+  const { segmentProps, segmentData } = props;
+  const { segmentDataState, segmentDataDispatch } = useSegmentContext();
+
+  const [conditionWidth, setConditionWidth] = useState(0);
+  const [filterOptionData, filterOptionDataDispatch] = useReducer(
+    analyticsSegmentFilterReducer,
+    {
+      ...INIT_SEGMENTATION_DATA,
+      conditionOptions: segmentDataState.attributeOptions,
+    }
+  );
+
+  const isDoneEvent = useMemo(() => {
+    return (
+      segmentData.userEventType?.value === ConditionType.USER_DONE ||
+      segmentData.userEventType?.value === ConditionType.USER_NOT_DONE
+    );
+  }, [segmentData.userEventType?.value]);
+
+  const isUserIsEvent = useMemo(() => {
+    return (
+      segmentData.userEventType?.value === ConditionType.USER_IS ||
+      segmentData.userEventType?.value === ConditionType.USER_IS_NOT
+    );
+  }, [segmentData.userEventType?.value]);
+
+  const isDoneInSeqEvent = useMemo(() => {
+    return (
+      segmentData.userEventType?.value === ConditionType.USER_DONE_IN_SEQUENCE
+    );
+  }, [segmentData.userEventType?.value]);
+
+  const isUserInGroup = useMemo(() => {
+    return (
+      segmentData.userEventType?.value === ConditionType.USER_IN_GROUP ||
+      segmentData.userEventType?.value === ConditionType.USER_NOT_IN_GROUP
+    );
+  }, [segmentData.userEventType?.value]);
```

Comment:
When `segmentData.userEventType?.value` has changed, all these four variables will get updated as well. I was wondering what we can benefit from using `useMemo()`. How about define a `userEventType` in state and update it via `useEffect()`?

Reply:
Good idea. updated it with useEffect